providers/elasticsearch/src/airflow/providers/elasticsearch/hooks/elasticsearch.py

```diff
@@ -235,20 +241,28 @@ class ElasticsearchPythonHook(BaseHook):
     """
     Interacts with Elasticsearch. This hook uses the official Elasticsearch Python Client.
 
+    .. deprecated:: 2.10.0
+        This hook is deprecated. Use :class:`~airflow.providers.elasticsearch.hooks.elasticsearch.ElasticsearchHook` instead.
+
     :param hosts: list: A list of a single or many Elasticsearch instances. Example: ["http://localhost:9200"]
     :param es_conn_args: dict: Additional arguments you might need to enter to connect to Elasticsearch.
                                 Example: {"ca_cert":"/path/to/cert", "basic_auth": "(user, pass)"}
     """
 
     def __init__(self, hosts: list[Any], es_conn_args: dict | None = None):
```

Comment:
Optional: add `-> None` to make the return type explicit.

```suggestion
    def __init__(self, hosts: list[Any], es_conn_args: dict | None = None) -> None:
```